lua/nvim-treesitter/parsers.lua

```diff
@@ -1748,6 +1748,16 @@ list.requirements = {
   readme_name = "pip requirements",
 }
 
+list.rescript = {
+  install_info = {
+    url = "https://github.com/rescript-lang/tree-sitter-rescript",
+    files = { "src/parser.c", "src/scanner.c" },
+    requires_generate_from_grammar = true,
+    use_makefile = true, -- macOS specific instruction
```

Comment:
Sorry, but that is not acceptable (anymore). If it doesn't build out of the box, it needs its own plugin.

Reply:
I don't think that's required, though -- but they do need to fix the include header for their scanner (and probably could use a bunch of other updates @amaanq)

Reply:
There is a PR to add all the generated files (and maybe it also fixes what you are talking about?). I'll mark as draft  and wait for that to get merged, that way no `generate_from_grammar` needed either

Reply:
The `generate_from_grammar` is fine; it's the `use_makefile` that is unacceptable.

(And the PR does not fix the build issue since it doesn't touch the scanner.)

I would recommend making a parallel PR _just_ fixing that (and maybe committing the two json files and headers); I have total sympathy for maintainers noping out on a +200K diffstat PR. (The bindings were a mistake. `ts g --no-bindings` 4tw.)

Reply:
Sounds good. Also is it possible the build issue is mac only? For me the parser builds when checking out this PR, and `ts g --no-bindings && ts t` works fine

Reply:
Could be platform (compiler) specific, yes. But it still should be fixed sooner this would pick up a system treesitter header (which could be any old version).